util/topsql/topsql.go

```diff
@@ -107,7 +115,31 @@ func AttachSQLInfo(ctx context.Context, normalizedSQL string, sqlDigest *parser.
 			sqlPrefixes := []string{"insert", "update", "delete", "load", "replace", "select", "begin",
 				"commit", "analyze", "explain", "trace", "create", "set global"}
 			if MockHighCPULoad(normalizedSQL, sqlPrefixes, 1) {
-				logutil.BgLogger().Info("attach SQL info", zap.String("sql", normalizedSQL), zap.Bool("has-plan", len(normalizedPlan) > 0))
+				logutil.BgLogger().Info("attach SQL info", zap.String("sql", normalizedSQL))
+			}
+		}
+	})
+	return ctx
+}
+
+// AttachSQLAndPlanInfo attach the sql and plan information into Top SQL
+func AttachSQLAndPlanInfo(ctx context.Context, sqlDigest *parser.Digest, planDigest *parser.Digest) context.Context {
```

Comment:
Why `linkSQLTextWithDigest` is called in `AttachSQLInfo` but is not called in `AttachSQLAndPlanInfo`?

Reply:
Maybe rename the function `AttachSQLInfo ` to `AttachAndRegisterSQLInfo` is better.

Since when topsql is disabled, when calling `AttachSQLAndPlanInfo` but not register SQL/Plan info for getting better performance.